Tasks/UniversalPackagesV0/universalpublish.ts

```diff
@@ -141,7 +111,44 @@ export async function run(artifactToolPath: string): Promise<void> {
             packageVersion: version,
         } as artifactToolRunner.IArtifactToolOptions;
 
-        publishPackageUsingArtifactTool(publishDir, publishOptions, toolRunnerOptions);
+        publishStatus = publishPackageUsingArtifactTool(publishDir, publishOptions, toolRunnerOptions);
+
+        var retries = 0;
+        let newVersion: string;
+
+        //If package already exist, retry with a newer version, do not retry for custom version
+        while (retries < numRetries && publishStatus != null && publishStatus == packageAlreadyExistsError && versionRadio != "custom") {
+            [serviceUri, packageName, feedId, projectId, accessToken] = authSetup(feedType);
+            if (feedType == "internal") {
```

Comment:
nit: Consider using [strict](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness#strict_equality_using) triple equality in comparisons, unless you have an explicit reason to use loose. The [rule to use only triple equal](https://github.com/microsoft/azure-pipelines-tasks/blob/master/tslint.json#L182) is turned on for this project. I also may recommend to use TSLint for your editor to see suggestions around the code :)

Reply:
Thanks for the input! For this case, should i change != into !== and == into ===?

The three equality comparison are for null check, int and string comparison.

Reply:
I would say that preferable to always use `===` instead of `==` for any non-null datatypes unless you have a justification (but in TS you usually not have a case for that, because you always are aware of the types you working with).
For the `publishStatus != null`. Because JS has two types that basically represents non-existing value (**undefined** and **null**), `!= null` used when you need  to check if the value is not equal to any of those, because:
```javascript
null == undefined // true
null === undefined // false
// so as follows
null != null // false
undefined != null // false
```
That is a justification why `!= null` [is allowed by TSLint rules](https://github.com/microsoft/azure-pipelines-tasks/blob/master/tslint.json#L184). In existence check I personally prefer to use `!= null`, because some people prefer to do `return` and some other prefer `return null` to exit from function without value and I have to be ready for both cases :). But if you want to use "not-equal" with any other datatypes, it is preferred to use `!==`.

Reply:
Fixed!

Thanks for the information, good to know about these rules.